openwisp_controller/subnet_division/rule_types/base.py

```diff
@@ -164,14 +164,14 @@ def get_config(cls, instance):
 
     @staticmethod
     def get_max_subnet(master_subnet, division_rule):
+        qs = Subnet.objects.filter(master_subnet_id=master_subnet.id)
+        if connection.vendor == 'postgresql':
+            qs = qs.order_by('-subnet')
+        else:
+            qs = qs.order_by('-created')
         try:
-            max_subnet = (
-                # Get the highest subnet created for this master_subnet
-                Subnet.objects.filter(master_subnet_id=master_subnet.id)
-                .order_by('-created')
-                .first()
-                .subnet
-            )
+            # Get the highest subnet created for this master_subnet
+            max_subnet = qs.first().subnet
```

Comment:
I suggest opting for something more compact and less lines changes:

```suggestion
            # <Add brief comment here>
            order_field = '-subnet' if connection.vendor == 'postgresql' else '-created'
            max_subnet = (
                # Get the highest subnet created for this master_subnet
                Subnet.objects.filter(master_subnet_id=master_subnet.id)
                .order_by(order_field)
                .first()
                .subnet
            )
```